### PR TITLE
Add Dockerfile and change config generation.

### PR DESCRIPTION
**Pull request by pragmader**
Now you can create Docker images for the website and specify all config
parameters as environment variables using `make config`. It's much
better for running in the cloud.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+public
+node_modules
+*.md
+Dockerfile
+*~
+.*
+*.log
+ehthumbs.db
+Thumbs.db
+__BrowserBundle.js
```
Comment (JuliaRechkunova) [inline .dockerignore]: Please add `.DS_Store`

File: README.md [hunk 2]
```diff
@@ -54,6 +67,18 @@ Also you can start it using [PM2](https://github.com/Unitech/PM2) like this:
...
+Or you can build a docker image:
+
+```bash
+docker build -t catberry/catberry-homepage .
+```
+
+And run it like this:
+
+```bash
+docker run -it -e CAT_PORT='3000' -e CAT_UA_ID='YOUR_GOOGLE_ANALYTICS' -e CAT_LOG_LEVEL='30' -e CAT_GITHUB_KEY='YOUR_GITHUB_APP_KEY_HERE' catberry/catberry-homepage
```
Comment (JuliaRechkunova) [inline README.md]: It does not work on Mac without `-p 3000:3000`

File: Makefile [hunk 3]
```diff
@@ -1,3 +1,9 @@
+SHELL := /bin/bash
+
+CONFIGS_DIR=./configs
+CONFIGS_EXAMPLE_DIR=./configs_example
+ENV_CONFIGURATION=$$(env | grep -oP '(.+)=' | grep -oP '^CAT_[\w\d]+')
```
Comment (JuliaRechkunova) [inline Makefile]: It's better to add some default values for env variables. At least for `CAT_LOG_LEVEL` and `CAT_PORT`.

File: .dockerignore (hunk 1 above)
Comment (pragmader) [inline .dockerignore]: It's not needed, we have `.*`

File: Makefile (hunk 3 above)
Comment (pragmader) [inline Makefile]: I like the idea, will do.

Comment (pragmader) [inline Makefile]: It seems to be a non-trivial problem, let's postpone it for a while.

**Pull request merged**